diff --git a/crypto/fipsmodule/rand/fork_detect_test.cc b/crypto/fipsmodule/rand/fork_detect_test.cc
index ab48512..cae3cb5 100644
--- a/crypto/fipsmodule/rand/fork_detect_test.cc
+++ b/crypto/fipsmodule/rand/fork_detect_test.cc
@@ -12,6 +12,7 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
 
+#include <cinttypes>
 #include <openssl/base.h>
 
 // TSAN cannot cope with this test and complains that "starting new threads
diff --git a/ssl/test/bssl_shim.cc b/ssl/test/bssl_shim.cc
index 62db076..86fa548 100644
--- a/ssl/test/bssl_shim.cc
+++ b/ssl/test/bssl_shim.cc
@@ -12,6 +12,7 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
 
+#include <cinttypes>
 #include <openssl/base.h>
 
 #if !defined(OPENSSL_WINDOWS)
diff --git a/util/embed_test_data.go b/util/embed_test_data.go
index 5376fdd49..a990bb3b9 100644
--- a/util/embed_test_data.go
+++ b/util/embed_test_data.go
@@ -90,6 +90,7 @@ func main() {
 
 #include <algorithm>
 #include <string>
+#include <stdio.h>
 

 `)
diff --git a/tool/internal.h b/tool/internal.h
index b626270f1..0834adea2 100644
--- a/tool/internal.h
+++ b/tool/internal.h
@@ -16,6 +16,7 @@
 #define OPENSSL_HEADER_TOOL_INTERNAL_H
 
 #include <openssl/base.h>
 #include <openssl/span.h>
+#include <stdio.h>
 
 #include <string>
 #include <utility>
