diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
index 64806d2ed..e7b9da363 100644
--- a/include/openssl/ssl.h
+++ b/include/openssl/ssl.h
@@ -4856,6 +4856,11 @@ OPENSSL_EXPORT int SSL_CREDENTIAL_get_ex_new_index(long argl, void *argp,
                                                    CRYPTO_EX_free *free_func);
 
 
+OPENSSL_EXPORT EVP_AEAD_CTX *SSL_get_write_aead_ctx(SSL *ssl);
+OPENSSL_EXPORT const uint8_t *SSL_get_write_iv_sequence(SSL *ssl,
+                                                        uint64_t **sequence,
+                                                        char *noad);
+
 // Low-level record-layer state.
 
 // SSL_get_ivs sets `*out_iv_len` to the length of the IVs for the ciphers
diff --git a/ssl/internal.h b/ssl/internal.h
index 74c6c2e9c..676af0e14 100644
--- a/ssl/internal.h
+++ b/ssl/internal.h
@@ -561,6 +561,10 @@ class SSLAEADContext {
 
   bool GetIV(const uint8_t **out_iv, size_t *out_iv_len) const;
 
+  EVP_AEAD_CTX *Get_evp_aead_ctx() { return ctx_.get(); }
+  const uint8_t *Get_fixed_nonce() const { return fixed_nonce_.data(); }
+  bool is_omit_ad() const { return false; }
+
  private:
   // GetAdditionalData returns the additional data, writing into `storage` if
   // necessary.
diff --git a/ssl/ssl_lib.cc b/ssl/ssl_lib.cc
index e05f91bd9..2d0cdc441 100644
--- a/ssl/ssl_lib.cc
+++ b/ssl/ssl_lib.cc
@@ -3052,6 +3052,19 @@ void SSL_set_renegotiate_mode(SSL *ssl, enum ssl_renegotiate_mode_t mode) {
   ssl_maybe_shed_handshake_config(ssl_impl);
 }
 
+EVP_AEAD_CTX *SSL_get_write_aead_ctx(SSL *ssl) {
+  auto *ssl_impl = FromOpaque(ssl);
+  return ssl_impl->s3->aead_write_ctx.get()->Get_evp_aead_ctx();
+}
+
+const uint8_t *SSL_get_write_iv_sequence(SSL *ssl, uint64_t **sequence, char *noad) {
+  auto *ssl_impl = FromOpaque(ssl);
+  *noad = ssl_impl->s3->aead_write_ctx.get()->is_omit_ad();
+  *sequence = &ssl_impl->s3->write_sequence;
+  return ssl_impl->s3->aead_write_ctx.get()->Get_fixed_nonce();
+}
+
+
 int SSL_get_ivs(const SSL *ssl, const uint8_t **out_read_iv,
                 const uint8_t **out_write_iv, size_t *out_iv_len) {
   // No cipher suites maintain stateful internal IVs in DTLS. It would not be

